File: advall-python/d3.py
numOfTriangles = 0
lst = []

def isTriangle(s1, s2, s3):
    return ((s1 + s2) > s3) and ((s1 + s3) > s2) and ((s2 + s3) > s1)


for line in open ("d3_input.txt", "r"):
    s = line.rstrip()
    t = [int(i) for i in s.split()]
    lst.append(t)
    if isTriangle(t[0], t[1], t[2]):
        numOfTriangles += 1

print "Part one: " + str(numOfTriangles)

numOfTriangles = 0

for i in xrange(0, len(lst), 3):
    for j in range(3):
        if isTriangle(lst[i][j], lst[i+1][j], lst[i+2][j]):
            numOfTriangles += 1

print "Part two: " + str(numOfTriangles)

File: advall-haskell/d3.hs


isTriangle :: [Integer] -> Bool
isTriangle [s1, s2, s3] = (s1 + s2 > s3) && (s1 + s3 > s2) && (s2 + s3 > s1)

countTriangles :: [[Integer]] -> Integer
countTriangles [] = 0
countTriangles (x:xs)
    | isTriangle x = 1 + countTriangles xs
    | otherwise = countTriangles xs

parseInput :: String -> [[Integer]]
parseInput s = map (map read) (map words (lines s))

parseInput' :: [[Integer]] -> [[Integer]]
parseInput' [] = []
parseInput' (x:y:z:xs) = [[x!!0] ++ [y!!0] ++ [z!!0]] ++ [[x!!1] ++ [y!!1] 
                ++ [z!!1]] ++ [[x!!2] ++ [y!!2] ++ [z!!2]] ++ parseInput' xs


main = do
    input <- readFile "d3_input.txt"
    let answer1 = countTriangles (parseInput input)
    let answer2 = countTriangles (parseInput' (parseInput input))
    putStrLn $ "Part one: " ++ show answer1
    putStrLn $ "Part two: " ++ show answer2